tests/protobuf-net.Grpc.Reflection.Test/ReflectionServiceTests.cs

```diff
@@ -0,0 +1,145 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Google.Protobuf.Reflection;
+using grpc.reflection.v1alpha;
+using ProtoBuf;
+using ProtoBuf.Grpc.Configuration;
+using ProtoBuf.Grpc.Reflection;
+using Xunit;
+
+namespace protobuf_net.Grpc.Reflection.Test
+{
+    using ReflectionTest;
+
+    public class ReflectionServiceTests
+    {
+        [Theory]
+        [MemberData(nameof(Dependencies))]
+        public async Task ShouldIncludeDependenciesInCorrectOrder(Type service, string symbolName, string[] expectedDescriptors)
+        {
+            IServerReflection reflectionService = new ReflectionService(service);
+
+            await foreach (var response in reflectionService.ServerReflectionInfoAsync(GetRequest()))
+            {
+                var fileDescriptors = response
+                    .FileDescriptorResponse
+                    .FileDescriptorProtoes
+                    .Select(x => Serializer.Deserialize<FileDescriptorProto>(x.AsSpan()))
+                    .ToArray();
+
+                Assert.Equal(expectedDescriptors, fileDescriptors.Select(x => x.Name));
+            }
+
+            async IAsyncEnumerable<ServerReflectionRequest> GetRequest()
+            {
+                yield return new ServerReflectionRequest
+                {
+                    FileContainingSymbol = symbolName
+                };
+
+                await Task.CompletedTask;
+            }
+        }
+
+        public static IEnumerable<object[]> Dependencies => new[]
+        {
+            new object[]
+            {
+                typeof(BclService),
+                ".ReflectionTest.BclService",
+                new[]
+                {
+                    "ProtoBuf.Grpc.Internal.Empty.proto", // TODO: Maybe Google well-known should have correct name?
```

Comment:
`Empty` doesn't seem to use Google well-known.. Should it?

Reply:
yeah, that's just wrong; we should generate the well-known name

Reply:
Fixed by using SchemaGenerator